### ISSUE TITLE
Rule suggestion: Emit warning for continuation characters in multi-line strings

### ISSUE THREAD

**Issue by dosisod** [astral-sh/ruff#7359]
I recently came across the following bug/surprising behavior, and it doesn't seem to be picked up by Ruff, Black, Flake8, etc (to my knowledge). Basically, if you use a continuation character in a multi-line string, the continuation character will merge the 2 lines. This probably isn't what most people intended, they probably meant to include them verbatim. This is similar to W605 (invalid escape sequence), except that this is perfectly valid, just not what most people expect.

For example:

```python
some_code = """
x = 1 \
    + 2 \
    + 3
"""
```

This is actually printed out like so:

```
x = 1     + 2     + 3
```

In short, Ruff should detect continuation characters inside multi-line strings and suggest/autofix it by turning the multi-line string into a raw string (similar to W605).

Cases where this shouldn't be detected are at the very start or end of a string, since this is usually indented:

```python
# ok
some_code = """\
print("hello world")
"""

# also ok
some_code = """
print("hello world") \
"""
```

Any thoughts? Thanks!
